homeassistant/components/weatherflow/config_flow.py

```diff
@@ -0,0 +1,69 @@
+"""Config flow for WeatherFlow."""
+from __future__ import annotations
+
+import asyncio
+from asyncio import Future
+from asyncio.exceptions import CancelledError
+from typing import Any
+
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.errors import AddressInUseError, EndpointError, ListenerError
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+async def _async_can_discover_devices() -> bool:
+    """Return if there are devices that can be discovered."""
+    future_event: Future[None] = asyncio.get_running_loop().create_future()
+
+    @callback
+    def _async_found(_):
+        """Handle a discovered device - only need to do this once so."""
+
+        if not future_event.done():
+            future_event.set_result(None)
+
+    async with WeatherFlowListener() as client, asyncio.timeout(10):
+        try:
+            client.on(EVENT_DEVICE_DISCOVERED, _async_found)
+            await future_event
+        except asyncio.TimeoutError:
+            return False
```

Comment:
```suggestion
            return
```

Reply:
actually this doesn't seem right to remove the check since if it times out, nothing checks for False and its going to create an entry anyways even if it didn't see a packet